go/3_auth-approle-with-response-wrapping.go

```diff
@@ -19,44 +20,37 @@ func getSecretWithAppRole() (string, error) {
 		return "", fmt.Errorf("unable to initialize Vault client: %w", err)
 	}
 
-	// A combination of a Role ID and Secret ID is required to log in to Vault with an AppRole.
-	// The Secret ID is a value that needs to be protected, so instead of the app having knowledge of the secret ID directly,
-	// we have a trusted orchestrator (https://learn.hashicorp.com/tutorials/vault/secure-introduction?in=vault/app-integration#trusted-orchestrator)
-	// give the app access to a short-lived response-wrapping token (https://www.vaultproject.io/docs/concepts/response-wrapping).
-	// Read more at: https://learn.hashicorp.com/tutorials/vault/approle-best-practices?in=vault/auth-methods#secretid-delivery-best-practices
-	wrappingToken, err := ioutil.ReadFile("path/to/wrapping-token") // placed here by a trusted orchestrator
-	if err != nil {
-		return "", fmt.Errorf("unable to read file containing wrapping token: %w", err)
-	}
-
-	unwrappedToken, err := client.Logical().Unwrap(strings.TrimSuffix(string(wrappingToken), "\n"))
-	if err != nil {
-		// a good opportunity to alert, in case the one-time use wrapping token appears to have already been used
-		return "", fmt.Errorf("unable to unwrap token: %w", err)
-	}
-	secretID := unwrappedToken.Data["secret_id"]
-
-	// the role ID given to you by your administrator
+	// A combination of a Role ID and Secret ID is required to log in to Vault
+	// with an AppRole.
+	// First, let's get the role ID given to us by our Vault administrator.
 	roleID := os.Getenv("APPROLE_ROLE_ID")
 	if roleID == "" {
 		return "", fmt.Errorf("no role ID was provided in APPROLE_ROLE_ID env var")
 	}
 
-	// log in to Vault's AppRole auth method
-	params := map[string]interface{}{
-		"role_id":   roleID,
-		"secret_id": secretID,
+	// The Secret ID is a value that needs to be protected, so instead of the
+	// app having knowledge of the secret ID directly, we have a trusted orchestrator (https://learn.hashicorp.com/tutorials/vault/secure-introduction?in=vault/app-integration#trusted-orchestrator)
+	// give the app access to a short-lived response-wrapping token (https://www.vaultproject.io/docs/concepts/response-wrapping).
+	// Read more at: https://learn.hashicorp.com/tutorials/vault/approle-best-practices?in=vault/auth-methods#secretid-delivery-best-practices
+	secretID := &auth.SecretID{FromFile: "path/to/wrapping-token"}
+
+	appRoleAuth, err := auth.NewAppRoleAuth(
+		roleID,
+		secretID,
+		auth.WithWrappingToken(),
```

Comment:
Should there be a comment about what this parameter is? I'm assuming it's a login option, but I'm not immediately sure what this is from looking at it (whether it's optional, etc.)

Reply:
Good point, I'll add a little inline comment there.